src/app/services/widgetservice/widget.service.ts

```diff
@@ -0,0 +1,33 @@
+import {Injectable} from '@angular/core';
+import {WidgetUiMode} from "../../widget/WidgetUiMode";
+import {WeatherWidget} from "../../interfaces/weatherwidget";
+import {interval, startWith} from "rxjs";
+import {map, switchMap} from "rxjs/operators";
+import {WeatherService} from "../weatherservice/weather.service";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WidgetService {
+  constructor(private weatherService: WeatherService){}
+  serviceData(widget:WeatherWidget){
```

Comment:
getAllWidgetData(widgets) {
this.serviceData https://www.learnrxjs.io/learn-rxjs/operators/combination/merge
}